cmd/kubeadm/app/phases/uploadconfig/uploadconfig_test.go

```diff
@@ -19,37 +19,79 @@ package uploadconfig
 import (
 	"testing"
 
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	clientsetfake "k8s.io/client-go/kubernetes/fake"
+	core "k8s.io/client-go/testing"
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
 func TestUploadConfiguration(t *testing.T) {
 	tests := []struct {
-		name    string
-		cfg     *kubeadmapi.MasterConfiguration
-		wantErr bool
+		name           string
+		createErr      error
+		updateErr      error
+		idempotentTest bool
+		wantErr        bool
+		verifyResult   bool
 	}{
 		{
-			"basic validation with correct key",
-			&kubeadmapi.MasterConfiguration{
-				KubernetesVersion: "1.7.3",
-			},
-			false,
+			name:         "basic validation with correct key",
+			wantErr:      false,
+			verifyResult: true,
+		},
+		{
+			name:           "update existing should report no error",
+			idempotentTest: true,
+			wantErr:        false,
+			verifyResult:   true,
+		},
+		{
+			name:      "unexpected errors for create should be returned",
+			createErr: apierrors.NewUnauthorized(""),
+			wantErr:   true,
```

Comment:
Small marginal gain, but it might be worth changing `wantErr` from a bool to the actual error we expect to be returned, which we can check for equality. We certainly wouldn't want to be returning the wrong errors in these cases.